### ISSUE TITLE
Crash on 10.12 calling registryID at device creation

### ISSUE THREAD

**Issue by reicrof** [KhronosGroup/MoltenVK#491]
Hi,
I encounter a crash with MoltenVK at device creation on MacOS 10.12. I assume the crash would also happen on 10.11 as well. The crash happens in MVKDevice::initPhysicalDevice:1985 when calling the registryID on the MTLDevice. It seems to be due to the fact that registryID is only available on 10.13+ (https://developer.apple.com/documentation/metal/mtldevice/2915737-registryid?language=objc).

As expected, when I run my test on 10.13, it works fine.

Here is my stack 
```
0   CoreFoundation                      0x00007fff7c1bb57b __exceptionPreprocess + 171
1   libobjc.A.dylib                     0x00007fff914351da objc_exception_throw + 48
2   CoreFoundation                      0x00007fff7c23bf14 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
3   CoreFoundation                      0x00007fff7c12ec93 ___forwarding___ + 1059
4   CoreFoundation                      0x00007fff7c12e7e8 _CF_forwarding_prep_0 + 120
5   libMoltenVK.dylib                   0x0000000102bc5a9b _ZN9MVKDevice18initPhysicalDeviceEP17MVKPhysicalDevice + 235
6   libMoltenVK.dylib                   0x0000000102bc508d _ZN9MVKDeviceC2EP17MVKPhysicalDevicePK18VkDeviceCreateInfo + 909
7   libMoltenVK.dylib                   0x0000000102bc6055 _ZN9MVKDeviceC1EP17MVKPhysicalDevicePK18VkDeviceCreateInfo + 37
8   libMoltenVK.dylib                   0x0000000102b5f5bc vkCreateDevice + 92
```

Am I missing something ? If my assumptions are correct it means no one on 10.12 and 10.11 could create a device ?

**Comment by billhollings** [comment]
Fixed in PR #494. Please retest and close if fixed.

**Comment by reicrof** [comment]
Fix looks good ! Thanks
